### ISSUE TITLE
Not clearing the previous lines when switching between select choices on Mac

### ISSUE THREAD

**Issue by Burgestrand** [piotrmurach/tty-prompt#17]
OS: `Mac v10.11.5` (El Capitan)
Terminal: iTerm and Terminal
$TERM: `xterm-256color`
Ruby: `2.3.1p122`

```sh
[~/Projects/bug-report]$ bash --noprofile # making sure I dont have any funky settings
bash-3.2$ ruby test.rb
What is your choice (Use arrow keys, press Enter to select)
What is your choice
What is your choice
What is your choice First
Selected: First.
```

I switch between the choices using up and down arrow, and the line keeps repeating. I assume this is more of an error in TTY::Cursor, rather than TTY::Prompt, but I'm not sure where you want bug reports.

Source code:
```ruby
require "bundler/setup"
require "tty-prompt"

prompt = TTY::Prompt.new

selected = prompt.select "What is your choice" do |menu|
  menu.choice "First"
  menu.choice "Second"
end

puts "Selected: #{selected}."
```

Versions of things:
```
GEM
  remote: https://rubygems.org/
  specs:
    equatable (0.5.0)
    necromancer (0.3.0)
    pastel (0.6.1)
      equatable (~> 0.5.0)
      tty-color (~> 0.3.0)
    tty-color (0.3.0)
    tty-cursor (0.3.0)
    tty-platform (0.1.0)
    tty-prompt (0.6.0)
      necromancer (~> 0.3.0)
      pastel (~> 0.6.0)
      tty-cursor (~> 0.3.0)
      tty-platform (~> 0.1.0)
      wisper (~> 1.6.1)
    wisper (1.6.1)

PLATFORMS
  ruby

DEPENDENCIES
  tty-prompt

BUNDLED WITH
   1.12.5
```

**Comment by piotrmurach** [comment]
Thanks for reporting this. It seems that El Capitan has a different interpretation for the ansi escape codes that move arrow up and/or clear the line. I agree, the problem is most likely in `tty-cursor` dependency. I don't run El Capitan so would really appreciate help here, otherwise I'm not sure how I would fix it as it works for me.

Have you run the specs? Maybe the failure shows the different escape codes being used?

Is this behaviour similar when you run `multi_select`? I would also try the `mask` and `slider` as they use similar line clearing ansi. If `mask` works as expected then it may imply that up/down arrow ansis are broken. Anyhow, you should be able to run quickly all the `examples`.

**Comment by piotrmurach** [comment]
One [change](https://github.com/piotrmurach/tty-cursor/commit/1ffa6beb3b2780bfa44812e9895eecc8c75e5984) that has been introduced in new `tty-cursor` was the code for moving to the previous line. [see](https://github.com/piotrmurach/tty-cursor/commit/1ffa6beb3b2780bfa44812e9895eecc8c75e5984). Therefore, would it possible for yourself to uninstall `tty-prompt` `v0.6.0` and install `v0.5.0` and see if `select` works there?

**Comment by Burgestrand** [comment]
Ok, neither `multi_select`, `mask` or `slider` clears data as expected. Here's what `mask` looks like, with an interesting error:

```
Are you on track for Launch Zipline by 2019-09-10? stty: 'standard input': unable to perform all requested operations
Are you on track for Launch Zipline by 2019-09-10? astty: 'standard input': unable to perform all requested operations
Are you on track for Launch Zipline by 2019-09-10? •sstty: 'standard input': unable to perform all requested operations
Are you on track for Launch Zipline by 2019-09-10? ••dstty: 'standard input': unable to perform all requested operations
Are you on track for Launch Zipline by 2019-09-10? •••astty: 'standard input': unable to perform all requested operations
```

Test suite for TTY::Prompt is all green!

**Comment by piotrmurach** [comment]
Hmmm.... the `mask` error makes me think that when `select` is called what may happen is that the error gets printed out to `sdterr` and then when the line is cleared it only clears the error and leaves the actual line to be cleared untouched? That would explain why the tests are passing as they are using `StringIO` to mimick `stdout` and `stderr`. Do you know where this error is coming from? It seems that unix `stty` utility tool is misbehaving?

**Comment by Burgestrand** [comment]
That sounds like a good theory. I'll dig.

**Comment by Burgestrand** [comment]
Yes, indeed, turns out that `%x[stty …]` for Echo and Raw sometimes prints an error to standard error (although I'm not sure when, or why). If I redirect stdout/stderr to `File::NULL` for all four of these commands, it appears to work properly.

```ruby
system("stty echo", $stdout => File::NULL, $stderr => File::NULL)
```

**Comment by piotrmurach** [comment]
Nice work, I suppose we could `%x{ stty ... >/dev/null 2>&1 }` and thus avoid shelling out with `system` command? Alternatively, we could try to use [io/console](http://ruby-doc.org/stdlib-2.0.0/libdoc/io/console/rdoc/IO.html#method-i-echo-3D) and latest `echo=` implementations? Would you have time to submit PR?

**Comment by Burgestrand** [comment]
Possibly today! Do you want to keep the separate Echo and Raw-classes? They might be superfluous if `io/console` is to be used, but since they've been declared public maybe we should keep them.

**Comment by piotrmurach** [comment]
I'm not really sure since when `io/console` is part of standard lib? Currently the code runs even on ruby `1.9.3`? I'm not precious about these classes and quite frankly would welcome removing them. Do you have time to look into this?

Another step after this fix is to extract the `Reader` to another dependency as it's usuful in its own way and provides conveniences that I can see easily applicable to other CLIs. 

Point taken re runing external commands, no matter how many times I use these I get them mixed up, that's probably why I've created [tty-command](https://github.com/piotrmurach/tty-command)

**Comment by Burgestrand** [comment]
I created a quick PR using the workaround code I used locally that uses `io/console`. If you decide that you'd prefer the `system` approach instead I'm happy to rewrite the PR; it's all just code. :)

**Comment by piotrmurach** [comment]
Released `v0.7.0`. Enjoy!
